app.js

```diff
@@ -1499,6 +1499,180 @@ app.delete("/room/:roomId", function (req, res) {
+          var suggestions = [];
+          //iterate through timetable and find first 5 suggestions
+          for(var time in timetable) {
+            if(countTimes == 5) {
+              break;
+            }
+            if(timetable[time] == 0) {
+              countTimes++;
+              suggestions.push(time);
+            }
+          }
+          //console.log(timetable);
+          //console.log(suggestions);
+          res.send([meetings, timetable, suggestions]);
+
+        }
+      });
+
+    }
+
+});
+
+function createTimetable(parameters) {
+
+    var meetings = parameters.meetings || null;
+    //var workdays = parameters.workdays || [1,2,3,4,5]; //list of workdays in the week (0-6)
+    var numDaysAhead = parameters.numDaysAhead || 3; //look ahead 3 days
+    var startTime = parameters.startTime || 7; //default start at 7AM
+    var endTime = parameters.endTime || 17; //default end at 5PM
+
+    //get current date info
+    var currDate = new Date();
+    var currHour = currDate.getHours();
+    var currMin = currDate.getMinutes();
+
+    currDate.setSeconds(0);
+    currDate.setMilliseconds(0);
+
+    //round up to the nearest 30 min
+    if(currMin > 30) {
+      currMin = 0;
+      currHour++;
+      currDate.setHours(currHour);
+      currDate.setMinutes(currMin);
+
+    }
+    else {
+      currMin = 30;
+      currDate.setHours(currHour);
+      currDate.setMinutes(currMin);
+    }
+
+    //if weekend, or friday after endTime, start searching monday at startTime
+    if(currDate.getDay() == 6 || currDate.getDay == 0 ||
+       (currDate.getDay() == 5 && currDate.getHour() >= endTime)) {
+      if(currDate.getDay() == 6) {
+        currDate.setDate(currDate.getDate() + 2);
+      }
+      else if(currDate.getDay() === 0) {
+        currDate.setDate(currDate.getDate() + 1);
+      }
+      else {
+        currDate.setDate(currDate.getDate() + 3);
+      }
+      currHour = startTime;
+      currMin = 0;
+      currDate.setHours(startTime);
+      currDate.setMinutes(0);
+    }
+
+    var timetable = {};
+
+    var workhours = endTime - startTime;
+    //search for the next 3 days
+    for(var i = 0; i < workhours*2*numDaysAhead; i++) { //*2 (for half hour intervals) * numDaysAhead to look
+      //if past endTime, go to startTime the next day
+      if(currHour >= endTime) {
+        currHour = startTime;
+        currMin = 0;
+        currDate.setHours(currHour);
+        currDate.setMinutes(currMin);
+        currDate.setDate(currDate.getDate()+1);
+      }
+
+      timetable[currDate.toUTCString()] = 0;
+
+      var x = 0;
+      //iterate through all meetings. if we find a meeting that conflicts, set that to busy
+      //if we don't, then the time is open/free
+      while(x < meetings.length) {
+        //i thought i'd be able to cut down on the meetings searched through, but this doesn't work???
+        // if((new Date(meetings[x].end_datetime)).getTime() > (new Date(currDate)).getTime()) {
```

Comment:
Here, i thought i could break out of this loop once I reach a point where the meeting date comes after the currDate, but this didn't seem to work

Reply:
???  

With the filter added to the meeting query you should only be scanning meetings that start in the future, so I don't think this check and break would be necessary (or do anything).